### DISCUSSION TITLE
Cosmograph 1.2: Visualizing Machine Learning Embeddings and more

### DISCUSSION THREAD

**Discussion by rokotyan** [cosmograph-org/.github#1] [Announcements]
We've got some exciting news about Cosmograph 🪐

If you don't know what Cosmograph is — it's the fastest web-based tool and library for visualizing large network graphs and now also two-dimensional machine learning embeddings.

![ui0-f8218d171556658274e76e8ec86deb63](https://github.com/cosmograph-org/.github/assets/755708/826cd34d-c495-41fb-9cf8-a05e903db017)

With the latest version of Cosmograph, you can:
1. Open a CSV file with a two-dimensional embedding of your data for visualization and analysis. 
2. Share your visualization with others by sending them a URL.
3. Integrate Cosmograph into your own web application (we have React and JavaScript packages).

In addition to all of the above, Cosmograph now has:
- Dynamic labels. As you zoom in or out, node labels will appear on the fly.
- Dedicated analysis tab with histograms for each numeric property.
- Improved search. You can search for a specific field and select multiple items found.

And there's a new website with detailed documentation and examples to make it easier for you to use Cosmograph and integrate it into your own tools. Give it a try and let us know what you think!

[https://cosmograph.app](https://cosmograph.app)

**Comment by oceanumeric** [comment]
This app is really great. Thanks a lot for making it as an open source! I love it so much and it helped to make sense of my network data with 0.5 - 1 million edges :)
